test/test_cuda.py

```diff
@@ -2999,7 +2999,7 @@ def test_autocast_rnn(self):
                                     device="cuda", dtype=hidden_dtype)
                     h = (h, c)
 
-                with torch.cuda.amp.autocast():
+                with torch.autocast():
```

Comment:
I think making `torch.autocast` default to cuda is suprising behavior, we should make `device` a required kwarg and not try to infer it.

Reply:
okay i will make that change